internal/painter/gl/gl_core.go

```diff
@@ -1,4 +1,4 @@
-// +build !gles,!arm,!arm64,!android,!ios,!mobile
```

Comment:
I think this change is not quite right.
If I have a linux arm64 this file should not be activated, but with that removed it will

Reply:
Yes, know this. Some options were just off. Rewrite all things with the following ideas in mind:

1. `*_core.go` files all getting `x OR (arm64 AND darwin)`, this means they are to be used with MacOS on Apple M1.
2. `*_es.go` files all getting `x AND (!arm64 AND darwin OR arm64 !darwin). This means they are excluded when compiling for MacOS on Apple M1.

Where `x` represents the existing set conditions.

BTW, `!arm64 AND darwin OR arm64 !darwin` == `!(arms64 AND darwin)`, this is a rule of boolean algebra. 
In fact I could simplify the negation in _es files for Apple M1 if keeping existing conditions in mind, but decided to stay it this way to keep things cleaner. You will be able to put another pairs if needed in inclusive conditions and append lines with exclusive conditions with just copy-pasting and minor editing.

Reply:
I think this has overcomplicated it now, sorry. the ES stuff should never run on darwin, and the core always should (as iOS is a different flag). All of the others were OK as far as I could see, just this one not.
Re-adding the "!arm64" to this line should and adding "arm64,darwin" at the end should have done it.

Reply:
I think the new change for this file is correct - but all of the other updates were not needed.
I'm not certain that your summary of boolean algebra is correct - you seem to have described exclusive or, which is not `!(x and y)`.

Reply:
> which is not !(x and y)

shit, indeed. Turned all conditions into `x AND !darwin` for _es and `x OR darwin` for _core

Reply:
Keep in mind though, this will work for Go > 1.16 only.

Reply:
I cannot see which elements of this change are only supporting Go 1.16?
We support back to 1.12, we will upgrade to 1.14 or 1.15 once Debian releases `testing`.

Reply:
`GOOS=darwin` for iOS prior Go 1.16. It is `GOOS=ios` since 1.16.

Reply:
Hmm, we embed the `gomobile` project for building `ios` and that sets the tag.
Does that mean that there is no problem?

Reply:
Looks like this indeed